source/extensions/tracers/xray/tracer.h

```diff
@@ -18,14 +21,18 @@ namespace Extensions {
 namespace Tracers {
 namespace XRay {
 
-static const char XRayTraceHeader[] = "x-amzn-trace-id";
+constexpr auto XRayTraceHeader = "x-amzn-trace-id";
 
 class Span : public Tracing::Span {
 public:
   /**
-   * Construct a new X-Ray Span.
+   * Creates a new Span.
+   *
+   * @param broker Facilitates communication with the X-Ray daemon.
+   * @param time_source A time source to create Span IDs using the monotonic clock.
    */
-  Span(TimeSource& time_source) : time_source_(time_source) {}
+  Span(Envoy::TimeSource& time_source, DaemonBroker* broker)
```

Comment:
Why not pass/store the broker as a reference? It should never be null.

Reply:
Yes. That's a left over from an earlier refactoring. Will change it. Thanks.